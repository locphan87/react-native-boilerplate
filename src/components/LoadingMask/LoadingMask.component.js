// @flow
import React from 'react'
import { ActivityIndicator } from 'react-native'

import { Wrapper } from '../Styled'
import type { GenericProps } from '../../types'

import styles from './LoadingMask.component.style'

const LoadingMask = (props: GenericProps) => (
  <Wrapper style={[styles.container, props.style]}>
    <ActivityIndicator />
  </Wrapper>
)
LoadingMask.displayName = 'LoadingMask'

export default LoadingMask
