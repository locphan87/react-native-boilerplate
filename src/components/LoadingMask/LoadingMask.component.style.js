import { StyleSheet } from 'react-native'

const styles = {
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0
  }
}

export default StyleSheet.create(styles)
