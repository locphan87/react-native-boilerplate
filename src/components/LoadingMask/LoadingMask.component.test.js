import { testSnapshots } from '../../utils/test.util'

import LoadingMask from './LoadingMask.component'

const props = {}

describe('LoadingMask', () => {
  testSnapshots(LoadingMask, [
    {
      props,
      description: 'basic render'
    }
  ])
})
