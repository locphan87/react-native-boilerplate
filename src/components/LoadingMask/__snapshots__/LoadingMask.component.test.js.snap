// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingMask snapshots basic render 1`] = `
<StyledWrapper
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</StyledWrapper>
`;
